nixpkgs_merge_bot/github/github_client.py

```diff
@@ -235,41 +235,51 @@ def merge_pull_request(
             log.debug(f"pull request {pr_number}: Staging, not merging")
             return None
 
-        # Using GraphQL's enablePullRequestAutoMerge mutation instead of the REST
-        # /merge endpoint, because the latter doesn't work with Merge Queues.
-        # This mutation works both with and without Merge Queues, and independently
-        # of whether the PR's required checks are still pending or have already
-        # succeeded.
-        # The only case where it doesn't work, is when there are no required status
-        # checks for the target branch. All development branches have these enabled,
-        # so this is a non-issue.
-        resp = self.post(
-            "/graphql",
-            data={
-                "query": dedent("""
-                    mutation ($node_id: ID!, $sha: GitObjectID) {
-                        enablePullRequestAutoMerge(input: {
-                            pullRequestId: $node_id,
-                            expectedHeadOid: $sha
-                        })
-                        {clientMutationId}
-                    }
-                """),
-                "variables": {"node_id": node_id, "sha": sha},
-            },
-        )
+        def graphql(mutation: str) -> HttpResponse:
```

Comment:
in the headmatter

`from typing import Literal`

```suggestion
        def graphql(mutation: Literal["enablePullRequestAutoMerge", "enqueuePullRequest", "mergePullRequest"]) -> HttpResponse:
```